lib/src/common/search_service.dart

```diff
@@ -47,7 +47,88 @@ class SearchService extends Service {
     return controller.stream;
   }
 
-  // TODO: Implement code: https://developer.github.com/v3/search/#search-code
+  /// Search through code for a given [query]
+  /// You can include any github qualifiers in the query directly
+  /// or you can set some of the optional params to set the qualifiers
+  /// For example, these do the same thing:
+  /// code('awesome language:dart') and
+  /// code('awesome', language: 'dart')
+  ///
+  /// https://developer.github.com/v3/search/#search-code
+  Future<SearchResults> code(String query,
+      {String language,
+      String filename,
+      String extension,
+      String user,
+      String org,
+      String repo,
+      String fork,
+      String path,
+      String size,
+      bool inFile: true,
+      bool inPath: false,
+      int pages: 2,
+      int perPage: 30}) async {
+    query += _searchQualifier('language', language);
+    query += _searchQualifier('filename', filename);
+    query += _searchQualifier('extension', extension);
+    query += _searchQualifier('user', user);
+    query += _searchQualifier('org', org);
+    query += _searchQualifier('repo', repo);
+    query += _searchQualifier('fork', fork);
+    query += _searchQualifier('path', path);
+    query += _searchQualifier('size', size);
+    String _in = '';
+    if (inFile) {
+      _in = 'file';
+    }
+    if (inPath) {
+      if (_in.isEmpty) {
+        _in = 'path';
+      } else {
+        _in = 'file,path';
+      }
+    }
+    if (_in.isNotEmpty) {
+      query += ' in:$_in';
+    }
+    var params = <String, dynamic>{};
+    params['q'] = query;
+    params['per_page'] = perPage?.toString();
+
+    var controller = new StreamController();
+
+    var isFirst = true;
+    SearchResults results = SearchResults();
+    new PaginationHelper(_github)
+        .fetchStreamed("GET", "/search/code", params: params, pages: pages)
+        .listen((response) {
+      if (response.statusCode == 403 &&
+          response.body.contains("rate limit") &&
```

Comment:
Do we really need to check that body? I feel like 403 alone would be a more generic catch that would work.